canvas/image.go

```diff
@@ -24,6 +24,16 @@ const (
 	ImageFillOriginal
 )
 
+// TextureFilter defines the different scaling filters used to scaling images
+type TextureFilter int32
```

Comment:
The word Filter tripped me up here - is it really a filter? The go docs seem to use the term "Interpolator" (which is quite ungainly).
The reason that I am unsure is that we are about to add FileFilter (or similar) to our file listing UI and the two different uses of the term could be confusing.
What would you think about "Sample" instead? That seems to be suggested higher up in google results than "filter" if I search for "image bilinear"... (open question for comment)

Reply:
Maybe we can use `ScalingMode`, same way as `FillMode` ?

Reply:
That's a good suggestion. Probably should keep the tense the same which would indicate `ScaleMode` instead of `ScalingMode`. That said I wonder if it clashes with the concept of scaling for size / position values. 
Anyone got thoughts?

Reply:
What about `InterpolationMode` with `Linear` and `NearestNeighbour` as currently possible values?

Reply:
GL call it Texture filter (`gl.TexParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR)`)

Reply:
Very true, but I think that doesn't really follow outside of GL and this is a toolkit global name

Reply:
so? ScalingMode ?

Reply:
Its long, but I'd go with TextureScalingFilter

Reply:
I'd drop Texture becauase that's a GL name.  It could be ImageScalingFilter I suppose. 
Looking again at the `Image.FillMode ImageFill` type it could indicate `image.ScaleMode ImageScale` with `ImageScale[Linear|NearestNeighbour]`.
It's a tough call: `Interpolation` seems correct but too complicated, `Filter` is clearly image related but in an instagram world means something else. `Scale` too has other connotations - all that is left from the suggestions is really `Sample` but that didn' seem popular.

I'll open a vote on the #fyne slack channel to see what our API consumers think.

Reply:
Possibly some food for thought from other environments:

iOS uses an InterpolationQuality setting which is another way to go
QT’s QImage uses a parameter to image.scale() such as Qt::FastTransformation
EFL seems to opt for “smooth_scale” bool
In general it seems like the implementation detail is hidden which is probably a good idea
And GDK-pixbuf does expose the algorithm choice under “GDK_INTERP_” prefix, so we wouldn’t be alone

Reply:
I wonder also about the values, perhaps they should mean more and/or not be implementation specific.
How about Pixel[Scale] and Smooth[Scale] perhaps? I guess there is also the "slow but best" option that is not being added in this PR.

Reply:
Looks like the channel has spoken and `ScaleMode` is the best of what we have just now.
So that's an indicator of what the field should be, then the field type would follow as `ImageScale` if we copy `FillMode`.
Then I would propose that we name the consts to match intent so perhaps `ImageScalePixels`, `ImageScaleSmooth`?